core/src/test/java/org/elasticsearch/index/SearchSlowLogTests.java

```diff
@@ -48,120 +50,93 @@ protected SearchContext createSearchContext(IndexService indexService) {
-                return new ShardSearchRequest() {
-                    @Override
-                    public ShardId shardId() {
-                        return new ShardId(indexService.index(), 0);
-                    }
-
-                    @Override
-                    public String[] types() {
-                        return new String[0];
-                    }
-
-                    @Override
-                    public SearchSourceBuilder source() {
-                        return null;
-                    }
-
-                    @Override
-                    public void source(SearchSourceBuilder source) {
-
-                    }
-
-                    @Override
-                    public int numberOfShards() {
-                        return 0;
-                    }
-
-                    @Override
-                    public SearchType searchType() {
-                        return null;
-                    }
-
-                    @Override
-                    public QueryBuilder filteringAliases() {
-                        return null;
-                    }
-
-                    @Override
-                    public long nowInMillis() {
-                        return 0;
-                    }
-
-                    @Override
-                    public Boolean requestCache() {
-                        return null;
-                    }
-
-                    @Override
-                    public Scroll scroll() {
-                        return null;
-                    }
-
-                    @Override
-                    public void setProfile(boolean profile) {
-
-                    }
-
-                    @Override
-                    public boolean isProfile() {
-                        return false;
-                    }
-
-                    @Override
-                    public BytesReference cacheKey() throws IOException {
-                        return null;
-                    }
-
-                    @Override
-                    public void rewrite(QueryShardContext context) throws IOException {
-                    }
-                };
+                return request;
             }
         };
     }
 
     public void testSlowLogSearchContextPrinterToLog() throws IOException {
         IndexService index = createIndex("foo");
-        // Turning off document logging doesn't log source[]
         SearchContext searchContext = createSearchContext(index);
-        SearchSlowLog.SlowLogSearchContextPrinter p = new SearchSlowLog.SlowLogSearchContextPrinter(searchContext, 10, true);
+        SearchSourceBuilder source = SearchSourceBuilder.searchSource().query(QueryBuilders.matchAllQuery());
+        searchContext.request().source(source);
+        SearchSlowLog.SlowLogSearchContextPrinter p = new SearchSlowLog.SlowLogSearchContextPrinter(searchContext, 10);
         assertThat(p.toString(), startsWith("[foo][0]"));
-    }
-
-    public void testReformatSetting() {
```

Comment:
funny, we were testing the setting and the test succeeded although the setting was ignored? :/